### PR TITLE
20.0 docs: unsupporting `gh-ost`

### PR DESCRIPTION
**Pull request by shlomi-noach**
Documenting the changes in https://github.com/vitessio/vitess/issues/15692

Unsupporting `gh-ost`:

- Pointing out that `gh-ost` is unsupported and will be removed.
- Replacing various `gh-ost` examples with `vitess` examples.
- Other long due updates to the docs.

### CODE REVIEW

File: content/en/docs/20.0/reference/errors/query-serving.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ These error messages are internal to Vitess. If you are getting other errors fro
 | 1192 | HY000 | Can't execute the given command because you have an active transaction | The provided statement cannot be executed inside a transaction. |
 | 1231 | 42000 | invalid transaction_mode: %s | Valid transaction_mode values are 'SINGLE', 'MULTI' or 'TWOPC'. |
 | 1231 | 42000 | invalid workload: %s | Valid workload values are 'OLTP', 'OLAP' or 'DBA'. |
-| 1231 | 42000 | invalid DDL strategy: %s | Valid DDL strategies are gh-ost, pt-osc. |
+| 1231 | 42000 | invalid DDL strategy: %s | Valid DDL strategies are `vitess`, `gh-ost`, `pt-osc`, `mysql`, `direct`. |
```
Comment (dbussink) [inline content/en/docs/20.0/reference/errors/query-serving.md:99]: Should we also update this in older docs versions? Looks like this was missed for older versions?

File: content/en/docs/20.0/reference/programs/vtctl/schema-version-permissions.md [hunk 2]
```diff
@@ -160,7 +160,7 @@ Applies the schema change to the specified keyspace on all shards. The recommend
 | Name | Type | Definition |
 | :-------- | :--------- | :--------- |
 | caller_id | string | Effective caller ID used for the operation and should map to an ACL name which grants this identity the necessary permissions to perform the operation (this is only necessary when strict table ACLs are used). |
-| ddl_strategy | string | Online DDL strategy, compatible with @@ddl_strategy session variable (examples: 'gh-ost', 'pt-osc', 'gh-ost --max-load=Threads_running=100' (default "direct"). |
+| ddl_strategy | string | Online DDL strategy, compatible with @@ddl_strategy session variable (examples: 'vitess', 'pt-osc', 'gh-ost --max-load=Threads_running=100' (default "direct"). |
```
Comment (dbussink) [inline content/en/docs/20.0/reference/programs/vtctl/schema-version-permissions.md:163]: Same here, this should probably also go into older docs versions? I think it applies to all command docs?

**Review by dbussink — APPROVED** [review]

File: content/en/docs/20.0/reference/programs/vtctl/schema-version-permissions.md (hunk 2 above)
Comment (shlomi-noach) [reply content/en/docs/20.0/reference/programs/vtctl/schema-version-permissions.md:163]: Done. It goes even beyond that, I'll review some of the other changes I made here which apply to versions 19, 18, 17.

File: content/en/docs/20.0/reference/errors/query-serving.md (hunk 1 above)
Comment (shlomi-noach) [reply content/en/docs/20.0/reference/errors/query-serving.md:99]: Done

**Comment by shlomi-noach** [comment]
Made final backport doc updates to `19.0`, `18.0`, `17.0`.

**Pull request merged**